src/components/java/patch-notes.tsx

```diff
@@ -75,33 +91,103 @@ async function PatchNotesImpl({
 
       if (!HElem) return;
 
-      const children = domNode.children as DOMNode[];
       const attribs = { ...domNode.attribs };
+      const children = domNode.children as DOMNode[];
+
       if (!attribs.id && children[0]?.type === ElementType.Text) {
-        attribs.id = toKebabCase(children[0].data);
+        const headingText: string = children.map(child =>
+          (child as DataNode)?.data ||
+          ((child as Element).children[0] as DataNode)?.data
```

Comment:
Perhaps there is a better way to write this, but in my experience `child` is either a `DataNode` or an `Element`, and the `as`'s were just to shut up the type errors 😅

I've tried out quite a few articles and haven't had any runtime errors, but if you have a suggestion for how to write it better I'll take it on board.

Also by the "null check" do you mean the optional chaining? There's no need for `?.` on `children` because it should as far as I can tell always exist if `data` doesn't.

Reply:
I mean the optional chaining in `(child as DataNode)?.data`, that's basically `child?.data` if types are ignored, and if `child` is null in the first branch, it will also be null in the second branch, so that optional chaining isn't helping with anything

Reply:
`child` will never be null, it's one of the children of the element. What might not exist is `data`, which is the purpose of the optional chaining, to return `undefined` instead of erroring if the key does not exist.

Reply:
well, the more you know. You'd think that getting a key that doesn't exist would error, but this *is* Javascript, so I don't know why I thought that :P